### ISSUE TITLE
[BUG]: RPI sees partitions from different disks as one disk

### ISSUE THREAD

**Issue by Igbarod** [raspberrypi/rpi-imager#960]
### What happened?

When I try to format the MicroSD with Raspberry Pi Imager, if I have Google Drive installed, it fails because it considers Google Drive a partition of the card. I also posted the issue in the [Raspberry forums](https://forums.raspberrypi.com/viewtopic.php?t=379181&sid=0a73de47b276f36e3a84227c9c035209) and the user incognitum gave some insight on the issue.

### Version

1.9.0

### What host operating system were you using?

Windows

### Host OS Version

22H2

### Selected OS

Raspberry Pi OS Bookworm 64-bit Lite

### Which Raspberry Pi Device are you using?

Raspberry Pi 4B, 400, and Compute Modules 4, 4S

### What kind of storage device are you using?

microSD Card in an internal reader

### OS Customisation

- [ ] Yes, I was using OS Customisation when the bug occurred.

**Comment by tdewey-rpi** [comment]
I agree, the current `drivelist` code leaves something to be desired - we've seen a range of interesting edge cases around virtual drives, software RAID and more exotic storage media - none of which present a good user experience.

I'm more minded to completely drop `drivelist`, and introduce a slimmer component that will perform an equivalent (hopefully sans undesired behaviours) function. This is aimed for Imager 2.0 at the earliest, unfortunately.
